compiler/vm/vm.nim

```diff
@@ -3180,4 +3180,123 @@ proc execute*(c: var TCtx, thread: var VmThread): YieldReason {.inline.} =
 template source*(c: TCtx, t: VmThread): TLineInfo =
   ## Gets the source-code information for the instruction the program counter
   ## of `t` currently points to
-  c.debug[t.pc]
\ No newline at end of file
+  c.debug[t.pc]
+
+
+func vmEventToAstDiagVmError*(evt: VmEvent): AstDiagVmError {.inline.} =
```

Comment:
This shouldn't be here, but because `vm` depends upon `ast_types` this is far less ceremony, for now.